### DISCUSSION TITLE
Tutorial and global vs. local position

### DISCUSSION THREAD

**Discussion by ffslight** [godotengine/godot-proposals#10636] [General Discussions]
I followed the tutorial at this step:
https://docs.godotengine.org/en/stable/getting_started/first_2d_game/05.the_main_game_scene.html#spawning-mobs
I recognize that i can have missed a step, but i was curious how the MobPath worked, so i tried to spawn entities without giving them any velocity, expecting them to form a rectangular pattern(follow the shape of my MobPath).
However they did not. And i dont even see them spawning on screen.
I then investigated and found that if i change their spawn location to use GlobalPosition:
mob.Position = mobSpawnLocation.GlobalPosition;

it would work as intended.

Because of this i figured it was an error in the tutorial and i submitted a pull request, to change it to global position, but it was rejected as the code works as intended. And i was suggested to start a discussion instead.

For reference this is my path:
![image](https://github.com/user-attachments/assets/29ff0f65-2d44-4414-bcb0-7256abbedb86)
and the mobspawnlocation:
![image](https://github.com/user-attachments/assets/d1bd4c7d-4935-4d31-91a5-8de4e8cf9de8)
Mobs are set to 0,0 in local space.

Here is an image side by side, with how the(in this case a placeholder) spawns randomly following the path using Globalposition as i suggest:
![image](https://github.com/user-attachments/assets/3ee49af5-5352-40f1-a804-58d5de64baa5)

So assuming i am doing something wrong since i can only make it work with GlobalPosition, what might that be?

Be aware that yes, the tutorial code "works", meaning that the game does as intended and spawns enemies randomly crossing the screen.

**Comment by AThousandShips** [comment]
This belongs [here](https://github.com/godotengine/godot-docs/issues) this is for feature proposals, so next time please open an issue there instead

**Comment by AThousandShips** [comment]
But suggestion something like:
```cs
mob.Position = mobSpawnLocation.GlobalPosition;
```
In this tutorial sends a very wrong message to new users, if anything it should be:
```cs
mob.GlobalPosition = mobSpawnLocation.GlobalPosition;
```

But regardless if the tutorial is followed correctly then the code works, and IMO we shouldn't introduce concepts that are beyond the scope of a tutorial like this, it makes things harder for beginners 

It would also *hide* errors from beginners, which would be a negative, instead we might want to add a note to remind people if they mess up to check their hierarchy

**Reply by ffslight** [reply]
You are right that setting both to GlobalPosition fixes things as well. I agree a note would have been helpful as well.
